linkerd/protocol/http/src/main/scala/io/buoyant/linkerd/protocol/http/ResponseFramingFilter.scala

```diff
@@ -0,0 +1,44 @@
+package io.buoyant.linkerd.protocol.http
+
+import com.twitter.finagle._
+import com.twitter.finagle.http.{Request, Response}
+import com.twitter.util.Future
+import io.buoyant.linkerd.protocol.FramingException
+
+/**
+ * A filter that fails badly-framed responses
+ */
+class ResponseFramingFilter extends SimpleFilter[Request, Response] {
+  override def apply(
+    request: Request,
+    service: Service[Request, Response]
+  ): Future[Response] =
+    service(request) flatMap { response =>
+      val numHeaders = response.headerMap.getAll("Content-Length").toSet.size
+      if (numHeaders > 1) {
+        // if the response has multiple Content-Length headers with unique
+        // values, it's invalid
+        Future.exception(FramingException(
```

Comment:
I think we should update our classifiers to be aware of this type of error.

How about we do this:
- Introduce a ProtocolError abstract type (in a protocol-agnostic module);
- Update existing classifiers to treat protocol errors like 5xx;
- Make the malformed framing filters throw a type that implements ProtocolError.

I think it's correct to treat this as an exception that should be handled in an error handler: in the router's stack, a 5xx from the remote service should be distinct from linkerd deciding it won't handle the response.